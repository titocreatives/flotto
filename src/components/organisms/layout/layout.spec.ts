import { shallowMount } from "@vue/test-utils";
import Layout from "./";

describe("Organism Layout", () => {
  let cmp: any;

  beforeAll(() => {
    cmp = shallowMount(Layout, {
      slots: {
        above: "above",
        underneath: "underneath"
      },
      stubs: ["router-view"]
    });
  });

  it("matches snapshot", () => {
    expect(cmp.element).toMatchSnapshot();
  });

  it("it renders slots", () => {
    expect(cmp.find(".layout-above").text()).toMatch("above");
    expect(cmp.find(".layout-underneath").text()).toMatch("underneath");
  });
});
