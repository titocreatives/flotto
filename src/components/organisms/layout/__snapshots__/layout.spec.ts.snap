// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organism Layout matches snapshot 1`] = `
<div
  class="o layout"
>
  <div
    class="layout-slide"
  >
    <header
      class="layout-above"
    >
      above
    </header>
     
    <section
      class="layout-content"
    >
      <router-view-stub
        name="fade"
      />
    </section>
     
    <div
      class="layout-underneath"
    >
      underneath
    </div>
  </div>
</div>
`;
